src/tequila/quantumchemistry/psi4_interface.py

```diff
@@ -548,3 +554,53 @@ def prepare_reference(self, *args, **kwargs):
             return super().prepare_reference(reference_orbitals=[i for i in range(len(active_reference_orbitals))],
                                              n_qubits=n_qubits, *args, **kwargs)
 
+    @property
+    def rdm1(self) -> tuple:
+        return super().rdm1
+
+    @property
+    def rdm2(self) -> tuple:
+        return super().rdm2
+
+    def compute_rdms(self, U: QCircuit = None, variables: Variables = None, spin_free: bool = True,
+                     get_rdm1: bool = True, get_rdm2: bool = True, psi4_rdms: bool = False):
+        """
+        Same functionality as qc_base.compute_rdms (look there for more information),
+        plus the additional option to compute 1- and 2-RDM using psi4 by the keyword psi4_rdms
+
+        Parameters
+        ----------
+        U :
+             Quantum Circuit to achieve the desired state \\psi = U |0\\rangle, optional if psi4_rdms is set to True
+        variables :
+            If U is parametrized, then need to hand over a set of fixed variables
+        spin_free :
+            Set whether matrices should be spin-free (summation over spin) or defined by spin-orbitals
+        get_rdm1, get_rdm2 :
+            Set whether either one or both rdm1, rdm2 should be computed. If both are needed at some point,
+            it is recommended to compute them at once.
+        psi4_rdms :
+            Set whether to compute matrices via psi4 or not.
+            If this is set to True, specifying U is no longer mandatory.
+
+        Returns
+        -------
+        """
+        if not psi4_rdms:
+            super().compute_rdms(U=U, variables=variables, spin_free=spin_free,
+                                 get_rdm1=get_rdm1, get_rdm2=get_rdm2)
+        else:
+            # Get 1- and 2-particle reduced density matrix via Psi4 CISD computation
+            cisd = self.compute_energy("fci", options={"detci__ex_level": 2,
```

Comment:
Can this be changed, so that we are more flexible.
Maybe the following:
Change psi4_rdms to psi4_method = None (None as default would mean it is not used)
additional keyword: psi4_options (default: None or empty dictionary)
Would make it more flexible so that one can use all CI and CC types which are able to compute rdms

In the current version there is one inconsitency: "FCI" and "detci__ex_level:2" are contradictions. ("detci" is the keyword for truncated CI calculations)

Reply:
Good that you mention detci (there used to be a problem with detci and active space, which seems to be resolved now). 
Generalizing to all CI types should be easily done, but obtaining them from e.g. CC or other methods using them internally does not seem to be straightforward (there's no function such as wavefunction.get_opdm() as for CIWavefunctions, or to rephrase it, the get_opdm/tpdm-functions only work for CIWavefunctions. Thus I could either add a comment that only CI-methods work right now, or try to dig around for a little bit longer (not sure if it'll pay off).

Reply:
Just add the comment. Having all truncated CI flavors is fine.